module/VuFind/src/VuFind/Controller/MyResearchController.php

```diff
@@ -2079,9 +2125,12 @@ public function newPasswordAction()
         }
         // Update hash to prevent reusing hash
         $this->getAuthManager()->updateUserVerifyHash($user);
-        // Login
+        if ($followUp = $this->followup()->retrieve('url')) {
+            $newUrl = strstr($followUp, 'Verify', true) . 'Home';
```

Comment:
I'm not sure I understand the purpose of this; maybe a comment is in order.

Reply:
added a comment. Happy to answer any other questions about this or taking ANY suggestions. Image is before this code block.

![image](https://github.com/user-attachments/assets/0810c1c5-9879-4652-8ff9-665de2239daf)

Reply:
Thanks, that makes sense, though I'm not sure I understand exactly what `$followUpUrl = strstr($followUp, 'Verify', true) . 'Home';` is trying to do. I wonder if it might be better to do something like `$followUpUrl = str_contains($followUp, 'Verify') ? $this->url()->fromRoute('home') : $followUp;` (so we're using the router instead of string manipulation to pick the destination URL... but maybe I'm misunderstanding which target route is actually desired).